api/db/populate-events.js

```diff
@@ -0,0 +1,97 @@
+const txtgen = require('txtgen');
+const { 
+  createPrompt,
+  getRandomDate,
+  getRandomCountry,
+  getRandomRegion ,
+  getRandomInterests
+} = require('./helper-functions');
+
+const Interest = require('../models/Interest');
+const User = require('../models/User');
+const Event = require('../models/Event');
+const { uniqueNamesGenerator, adjectives, animals, starWars, colors } = require('unique-names-generator');
+
+const createEvent = async (eventParams) => {
+  const event = new Event({
+    name: eventParams.name,
+    description: eventParams.description,
+    startDate: eventParams.startDate,
+    endDate: eventParams.endDate,
+    country: eventParams.country,
+    region: eventParams.region,
+    createdBy: eventParams.user,
+    relatedInterests: eventParams.interests,
+    coverPhoto: eventParams.coverPhoto,
+    attendantsLimit: eventParams.attendantsLimit,
+  });
+  await event.save();
+
+  if (displayLogs) {
+    console.log('\n', '\x1b[0m', `New event created: ${event}`);
+  }
+};
+
+const createEvents = async (multiplier, randomLocation, verbose) => {
+  console.log('\n', '\x1b[0m', 'Populating events collection...');
+
+  let usersList = await User.distinct('_id');
+  if (usersList.length === 0) {
+    console.log(
+      '\x1b[33m',
+      'The users collection does not exist. Aborted events collection population.'
+    );
+    return;
+  }
+
+  let interestsList = await Interest.distinct('_id');
+  if (interestsList.length === 0) {
+    const proceed = await createPrompt(
+      'The interests collection does not exist. Events created will not have any related interests. Do you want to continue?'
+    );
+    if (!proceed) {
+      console.log('\x1b[33m', 'Aborted events collection population');
+      return;
+    }
+  }
+
+  displayLogs = verbose;
+  const eventsArray = [];
+  const now = new Date();
+  const futureDate = new Date().setFullYear(now.getFullYear() + 5);
+
+  for (let i = 0; i < 500 * multiplier; i++) {
+    const name = uniqueNamesGenerator({
+      dictionaries: [adjectives, animals, colors, starWars],
+      length: 3,
+      separator: ' '
+    });
+
+    const description = txtgen.paragraph();
```

Comment:
Maybe we could use the `article()` method with a random number of paragraphs here so we can have event descriptons with different lengths, which will allow us to check that everything looks great regardless of the size of them.

Reply:
What is the longest a description can be? If you didn't provide one, there may be a restriction with mongodb that I can look up.

Reply:
Just looked up mongodb's restrictions. The largest size a document can be is 16 MB, so we should be good.